### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by ruslanshakirov**

### CODE REVIEW

File: app/controllers/stories_controller.rb [hunk 1]
```diff
@@ -31,7 +31,9 @@ def update
   end
 
   def show
-    @reviews = story.reviews.includes(:user).order("created_at").decorate
+    if story.reviews.present?
+      @reviews = story.reviews.includes(:user).order("created_at").decorate
```
Comment (AlinaMoskieva) [inline app/controllers/stories_controller.rb]: @reviews = story.reviews.includes(:user).order("created_at").decorate if story.reviews.present?

Comment (ruslanshakirov) [inline app/controllers/stories_controller.rb]: @AlinaMoskieva line will be long

File: spec/features/writer/publish_story_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  scenario "Writer creates story with name" do
+    visit new_story_path
+    find("input[placeholder='Set name of your story here']").set "Three Comrades"
+    click_button "Publish"
+    expect(page).to have_text "Three Comrades"
+    expect(page).to have_link "Add profile"
+    expect(page).to have_link "Add text"
+  end
+
```
Comment (AlinaMoskieva) [inline spec/features/writer/publish_story_spec.rb]: let(:story_name) { "Three Comrades" }

File: app/controllers/stories_controller.rb [hunk 3]
```diff
@@ -31,6 +32,7 @@ def update
   end
 
   def show
+    return unless story.reviews.present?
     @reviews = story.reviews.includes(:user).order("created_at").decorate
```
Comment (vladimirbazhanov) [inline app/controllers/stories_controller.rb]: This action can be a problem in future.

Look, if there are any reviews - you will get a collection
If there are no reviews - you will get nil

So you must remember you can receive nil in your view and do a check. It's much better if you will return collection in both situations, but if there is no review - return empty collection. not nil

You can google 'Null object' to learn more about this

File: app/decorators/review_decorator.rb [hunk 4]
```diff
@@ -6,6 +6,10 @@ def date
   end
 
   def user_full_name
```
Comment (vladimirbazhanov) [inline app/decorators/review_decorator.rb]: It's better to use one of two variants

1. Or delete reviews of that user right after user removed. And you do not need to check if user exists
2. Or add 'is_deleted' flag for user and do not really remove user from DB

is other cases you will have user_id for not existed user - it very simular to broken database)

File: app/policies/story_policy.rb [hunk 5]
```diff
@@ -32,8 +32,8 @@ def text?
   end
 
   def filled_profile?
-    record.genre && record.full_description.present? &&
-      record.short_description.present? && record.cover.present?
+    record.genre || record.full_description.present? ||
+      record.short_description.present? || record.cover.present?
```
Comment (vladimirbazhanov) [inline app/policies/story_policy.rb]: (record.genre, record.full_description, record.short_description, record.cover).any?

File: app/views/stories/_profile_buttons.html.slim [hunk 6]
```diff
@@ -0,0 +1,10 @@
+.profile_buttons
+  - if policy(story).owner_or_admin?
+    - if !policy(story).filled_profile?
+      = link_to "Add profile", story_profile_index_path(story), class: "button small secondary button_item"
+    - else 
+      = link_to "Change profile", story_profile_index_path(story), class: "button small secondary button_item"
+    - if policy(story).text?
```
Comment (vladimirbazhanov) [inline app/views/stories/_profile_buttons.html.slim]: policy must answer only to CRUD action names, for example, create?, update?, edit?

.text? in Ruby mean that you want to ask object is it 'text' or it is not :) This is not understandable by other developers)

File: app/views/stories_list/_story_table.html.slim [hunk 7]
```diff
@@ -2,8 +2,8 @@ table
   thead
     tr
       th width="200" Name of story
-      th Author
-      th Created at
+      th width="150"Author
+      th width="150" Created at
```
Comment (vladimirbazhanov) [inline app/views/stories_list/_story_table.html.slim]: can this be moved to CSS?

File: spec/features/writer/add_profile_spec.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    click_link "Add profile"
+  end
+
+  scenario "Visit profile page" do
+    expect(page).to have_text "Add/change information about your story"
+    expect(page).to have_css "input#story_name"
+    expect(page).to have_selector(:css, "#story_genre")
+    expect(page).to have_css "textarea#story_short_description"
+    expect(page).to have_css "textarea#story_full_description"
+    expect(page).to have_css "input#story_cover"
```
Comment (vladimirbazhanov) [inline spec/features/writer/add_profile_spec.rb]: pls use one check per test.

Why to check all fields of form at page? Check only form tag and you will get the same result like 'form is there'. Or 'form is not there'

File: spec/features/writer/add_profile_spec.rb [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    expect(page).to have_button "Publish"
+    expect(page).to have_link "Delete"
```
Comment (vladimirbazhanov) [inline spec/features/writer/add_profile_spec.rb]: Checking for 'delete' link looks like testing of policy?

Its better to create other scenario and name it 'User can delete story' and that link presence. Or absence.

**Pull request merged**